pure_pursuit_wrapper/src/pure_pursuit_wrapper_node.cpp

```diff
@@ -28,14 +28,26 @@ int main(int argc, char** argv) {
   ros::NodeHandle nh("");
 
   pure_pursuit_wrapper::PurePursuitWrapper PurePursuitWrapper(nh);
+  ros::CARMANodeHandle::setSpinRate(10);
 
   // Approximate time of 100ms used because NDT outputs at 10Hz
   message_filters::Synchronizer<pure_pursuit_wrapper::PurePursuitWrapper::SyncPolicy> sync(pure_pursuit_wrapper::PurePursuitWrapper::SyncPolicy(100), PurePursuitWrapper.pose_sub, PurePursuitWrapper.trajectory_plan_sub);
   sync.registerCallback(boost::bind(&pure_pursuit_wrapper::PurePursuitWrapper::TrajectoryPlanPoseHandler, &PurePursuitWrapper, _1, _2));
 
-  while (ros::ok() && !PurePursuitWrapper.shutting_down_) {
-    ros::spinOnce();
-  }
+  //ros::Rate r(10);
+  //while (ros::ok() && !PurePursuitWrapper.shutting_down_) {
+  //  PurePursuitWrapper.PublishPluginDiscovery();
+  //  ros::spinOnce();
+  //  r.sleep();
+ // }
+
+  ros::CARMANodeHandle::setSpinCallback([this]() -> bool
+  {
+  if (!PurePursuitWrapper.shutting_down_)
+  {PurePursuitWrapper.PublishPluginDiscovery();}
+  return !PurePursuitWrapper.shutting_down_;
+  });
+  ros::CARMANodeHandle::spin();
 
   ROS_INFO("Successfully launched node.");
```

Comment:
This should probably before the `spin()`, else this'll only get printed when the system has finished spinning and is shutting down.